File: backend/app/models/notification.py
from datetime import datetime, timezone
from bson import ObjectId


def notification_doc(
    user_id: str,
    notification_type: str,
    title: str,
    message: str,
    metadata: dict | None = None,
) -> dict:
    """Build a notification document ready for insert_one()."""
    return {
        "user_id": ObjectId(user_id),
        "type": notification_type,       # "sync_success" | "sync_error" | "info"
        "title": title,
        "message": message,
        "read": False,
        "metadata": metadata or {},
        "created_at": datetime.now(timezone.utc),
    }

File: frontend/types/notification.ts
export type NotificationType =
  // ── Task workflow ─────────────────────────────────────────────────────────
  | "task_assigned"       // a task was assigned to you
  | "task_started"        // an employee started a task (→ leader)
  | "task_break"          // an employee paused a task (→ leader)
  | "pending_review"      // task submitted for review
  | "task_approved"       // task was approved
  | "task_reedit"         // task sent back for revision
  | "team_task_assigned"  // new task added to your team (→ leader)
  | "due_date_reminder"   // task is due tomorrow
  // ── Data sync ─────────────────────────────────────────────────────────────
  | "sync_success"
  | "sync_error"
  // ── Generic ───────────────────────────────────────────────────────────────
  | "info";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  read: boolean;
  metadata: Record<string, unknown>;
  created_at: string;
}

export interface NotificationsData {
  items: Notification[];
  unread_count: number;
}
